### ISSUE TITLE
bug: Nested collapse button doesn't change state when collapse-open classname is toggled 

### ISSUE THREAD

**Issue by krislander** [saadeghi/daisyui#2336]
### What version of daisyUI are you using?

3.6.3

### Describe your issue

```import { cn } from '@/lib/utils';
import { useState } from 'react';

export default function Expander({
  title,
  children,
  openByDefault,
  sticky = false,
  nested = false,
}: {
  title: React.ReactNode;
  children: React.ReactNode;
  openByDefault?: boolean;
  sticky?: boolean;
  nested?: boolean;
}) {
  const [expanded, setExpanded] = useState(openByDefault);

  return (
    <div
      className={cn(
        'collapse collapse-plus rounded',
        expanded && 'collapse-open overflow-visible',
        nested ? 'bg-base-200 border border-gray-300' : 'bg-base-100 '
      )}
    >
      <div
        onClick={(e) => {
          setExpanded((expanded) => !expanded);
        }}
        className={cn(
          'collapse-title text-lg cursor-pointer z-10 shadow',
          sticky && 'sticky top-0',
          nested ? 'bg-base-200 border border-gray-300' : 'bg-base-100 '
        )}
      >
        {title}
      </div>

      <div
        className={cn('collapse-content overflow-x-auto',
          nested
            ? 'border bg-base-300 border-x-gray-300 border-b-gray-300'
            : 'bg-base-200',
          expanded && 'pt-4')}
      >
        {children}
      </div>
    </div>
  );
}
```

The inner Expander must have nested passed as prop, to have the right styling. The sticky prop is to keep the collapse on top of the parent container until it is scrolled through. (not important for this case)
What I am trying to achieve here, is to be able to control the collapse-open property of both the parent and nested collapse. currently in this code chunk, I achieved the closest result to the expected one, but still, the collapse-plus of the child expander doesn't respond when trigerred. The collapse-content also seems to not fold fully when collapse is closed. This also occurs only in the child element. The pictures below show that the current open and close states look like for the parent and child.

1. All collapses open
![image](https://github.com/saadeghi/daisyui/assets/22302243/030c9d88-fe36-495c-b805-3b41c88deef3)

2. Parent collapse closed
![image](https://github.com/saadeghi/daisyui/assets/22302243/08d92756-44b9-40e1-9022-9e6603b6f809)

3. Nested collapse closed only
![image](https://github.com/saadeghi/daisyui/assets/22302243/938f4827-abf1-4c01-9c9c-65130ba9436e)

### What browsers are you seeing the problem on?

All browsers

**Comment by saadeghi** [comment]
Please update to the latest version. I think this is already fixed.  
If not, can you please provide compiled HTML code or put your component file on a StackBlitz or something so I can run it? because the code you provided is just one `collapse`. It's not nested.

**Comment by krislander** [comment]
> Please update to the latest version. I think this is already fixed. If not, can you please provide compiled HTML code or put your component file on a StackBlitz or something so I can run it? because the code you provided is just one `collapse`. It's not nested.

Thank you. This resolved the issue for me.

**Comment by krislander** [comment]
Fixed with version 3.7.3
